alfalfa_worker/lib/run_manager.py

```diff
@@ -123,83 +126,196 @@ def checkout_run(self, run_id: str) -> Run:
+        """Upload JSON documents to mongo.  The documents look as follows:
         {
             '_id': '...', # this maps to the 'id' below, the unique id of the entity record.
-            'site_ref': '...', # for easy finding of entities by site
             'rec': {
                 'id': '...',
                 'siteRef': '...'
                 ...other Haystack tags for rec
             }
         }
         :param haystack_json: json Haystack document
-        :param site_ref: id of site
-        :return: pymongo.results.InsertManyResult
+        :param run: id of the run
+        :return: None
         """
-        array_to_insert = []
-        for entity in haystack_json:
-            array_to_insert.append({
-                '_id': entity['id'].replace('r:', ''),
-                'site_ref': run.id,
-                'rec': entity
-            })
-        response = self.mongo_db_recs.insert_many(array_to_insert)
-        return response
+        # Create a site obj here to keep the variable active, but it is
+        # really set as index 0 of the haystack_json
+        site = None
+
+        # create all of the recs for this site
+        for index, entity in enumerate(haystack_json):
+            if index == 0:
+                # this is the site record, store it on the site object of the
+                # database (as well as in the recs collection, for now).
+                # TODO: convert to actual data types (which requires updating the mongo schema too)
+                # TODO: FMU's might not have this data?
+                name = f"{entity.get('dis','Test Case').replace('s:','')} in {entity.get('geoCity', 'Unknown City').replace('s:','')}"
+                site = Site(ref_id=run.ref_id, name=name).save()
+                site.haystack_raw = haystack_json
+                site.dis = entity.get('dis')
+                site.site = entity.get('site')
+                site.area = entity.get('area')
+                site.weather_ref = entity.get('weatherRef')
+                site.tz = entity.get('tz')
+                site.geo_city = entity.get('geoCity')
+                site.geo_state = entity.get('geoState')
+                site.geo_country = entity.get('geoCountry')
+                site.geo_coord = entity.get('geoCoord')
+                site.sim_status = entity.get('simStatus')
+                site.sim_type = entity.get('simType')
+                site.save()
+
+            rec = Rec(ref_id=entity['id'].replace('r:', ''), site=site).save()
+            rec_instance = RecInstance(**entity)
+            rec.rec = rec_instance
+            rec.save()
+
+            # Create default writearray objects for the site. Create this on the backend
+            # to ensure that the links are created correctly and accessible from the frontend.
+            #   Only check for records tagged with writable.
+            #   This may need to be expanded to other types in the future.
+            if rec.rec.writable == 'm:':
```

Comment:
this is where the writearrays are populated in mongo. No longer being created from the frontend. This will likely change to being just Redis???